client/src/components/EditContactPopup.jsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react'
+import Modal from 'react-bootstrap/Modal'
+import Button from 'react-bootstrap/Button'
+import { useState } from 'react';
+
+export default function EditContactPopup({show , onClose, contact, onContactChange}) {
+
+  const [editContact, setEditContact] = useState({
+    "first_name" : contact.person.first_name,
+    "last_name" : contact.person.last_name,
+    "email" : contact.person.email,
+    "phone_number": contact.person.phone_number,
+    "address": contact.person.address,
+    "image_path" : contact.person.image_path
+  })
+
+  function handleChange (event) {
+
+    const { value, name } = event.target;
+    setEditContact((prevValue) => ({ ...prevValue, [name]: value }));
+  }
+
+  const editTheContact= async (editContact) => {
+    try {
+
+      const {first_name, last_name, email, phone_number, address, image_path} = editContact;
+
+      const body = {first_name, last_name, email, phone_number, address, image_path};
```

Comment:
```suggestion
      const body = { ...editContact };
```

You can simplify this code with the [`...` (spread) operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax)